test/check.sh

```diff
@@ -570,6 +570,24 @@ it_can_check_with_tag_filter() {
   "
 }
 
+it_can_check_with_tag_regex() {
+  local repo=$(init_repo)
+  local ref1=$(make_commit $repo)
+  local ref2=$(make_annotated_tag $repo "1.0-staging" "tag 1")
+  local ref3=$(make_commit $repo)
+  local ref4=$(make_annotated_tag $repo "1.0-production" "tag 2")
+  local ref5=$(make_annotated_tag $repo "2.0-staging" "tag 3")
+  local ref6=$(make_commit $repo)
+  local ref7=$(make_annotated_tag $repo "2.0-staging" "tag 5")
+  local ref8=$(make_commit $repo)
+  local ref9=$(make_annotated_tag $repo "2.0-production" "tag 4")
+  local ref10=$(make_commit $repo)
+
+  check_uri_with_tag_regex $repo ".*-staging" | jq -e "
+    . == [{ref: \"2.0-staging\", commit: \"$ref6\"}]
```

Comment:
why in here the check doesn't return `ref3` and `ref8`? what is the reasoning behind `tail -1` in the implementation? Thx.

Reply:
hey @xtremerui, thanks for taking the time to review! :)

I have taken the approach of duplicating the exact same logic as for the `tag_filter` option, which also behaves this way (uses [`tail -1`](https://github.com/concourse/git-resource/blob/master/assets/check#L149)) in this scenario. I've also duplicated the exact same set of tests as for `tag_filter`, which in this case also return only `ref6`. 

I can't say for certain whether this is _correct_, but I can say that it's at least _consistent_ with `tag_filter` in terms of behaviour. 

I'd be more than happy to add additional tests here if you think that would clarify the situation.

Reply:
I think the check would only return the latest version by default so `tail -1` makes sense here. However `ref8` is actually newer than `ref6` plus it matches the regx, should `ref8` be expected other than `ref6`?

Reply:
correct me if I'm wrong, but `ref9`  (`2.0-production`) points to `ref8`. and `ref7` (`2.0-staging`) points to `ref6`. to me this looks correct, I would expect the regex `.*-staging` to return the latest commit that is pointed to by the `2.0-staging` tag, which is `ref6`. 

(side note: this is the same behaviour as the test directly above this one, on like 568)